app/services/timer-service.js

```diff
@@ -0,0 +1,43 @@
+import Ember from 'ember';
+
+export default Ember.Service.extend({
+  _countdown() {
+    if (this.get('seconds') === 0) {
+      this.set('timeUp', true);
+    } else {
+      this.incrementProperty('seconds', -1);
+      this._timer = Ember.run.later(this, '_countdown', 1000);
```

Comment:
When it reaches 0, the first conditional branch is followed and _countdown is never called again. That ends the recursion. #verified

On another note, timeUp currently isn't used for anything but it could be used for a UI state change.